ui/src/ScholarReader.tsx

```diff
@@ -754,6 +754,8 @@ export default class ScholarReader extends React.PureComponent<Props, State> {
             {} as { [s2Id: string]: Paper }
           );
           this.setState({ papers, areCitationsLoading: false });
+        } else {
```

Comment:
Is this really **3** levels of nested `if` statements? I suspect it would be worth breaking this function up into smaller functions...but maybe that should be done when adding a store abstraction.

Reply:
Well the top two levels should really just be 1...but yes, when we add stores we'll have opportunities to move this code out of the top component & break it up.